website/content/en/docs/kubebuilder/references/logging.md

```diff
@@ -0,0 +1,261 @@
+	logf.SetLogger(logger)
+
+	scopedLog := logf.Log.WithName("scoped")
+
+	globalLog.Info("Printing at INFO level")
+	globalLog.V(1).Info("Printing at DEBUG level")
+	scopedLog.Info("Printing at INFO level")
+	scopedLog.V(1).Info("Printing at DEBUG level")
+}
+```
+
+#### Output using the defaults
+```console
+$ go run main.go
+INFO[0000] Running the operator locally in namespace default. 
+{"level":"info","ts":1587741740.407766,"logger":"global","msg":"Printing at INFO level"}
+{"level":"info","ts":1587741740.407855,"logger":"scoped","msg":"Printing at INFO level"}
+```
+
+#### Output overriding the log level to 1 (debug)
+```console
+$ go run main.go --zap-log-level=debug
+INFO[0000] Running the operator locally in namespace default. 
+{"level":"info","ts":1587741837.602911,"logger":"global","msg":"Printing at INFO level"}
+{"level":"debug","ts":1587741837.602964,"logger":"global","msg":"Printing at DEBUG level"}
+{"level":"info","ts":1587741837.6029708,"logger":"scoped","msg":"Printing at INFO level"}
+{"level":"debug","ts":1587741837.602973,"logger":"scoped","msg":"Printing at DEBUG level"}
+```
+## Custom zap logger
+
+In order to use a custom zap logger, [`zap`][controller_runtime_zap] from controller-runtime can be utilized to wrap it in a `logr` implementation.
+
+Below is an example illustrating the use of [`zap-logfmt`][logfmt_repo] in logging.
+
+### Example
+
+In your `main.go` file, replace the current implementation for logs inside the `main` function:
+
+```Go
+...
+// Add the zap logger flag set to the CLI. The flag set must
+// be added before calling flag.Parse().
+	opts := zap.Options{}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	logger := zap.New(zap.UseFlagOptions(&opts))
+	logf.SetLogger(logger)
+...
+```
+
+With:
+
+```Go
+	import(
+	...
+	zaplogfmt "github.com/sykesm/zap-logfmt"
+	uzap "go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	...
+)
+	configLog := uzap.NewProductionEncoderConfig()
+	configLog.EncodeTime = func(ts time.Time, encoder zapcore.PrimitiveArrayEncoder) {
+		encoder.AppendString(ts.UTC().Format(time.RFC3339Nano))
+	}
+	logfmtEncoder := zaplogfmt.NewEncoder(configLog)
+
+	// Construct a new logr.logger.
+	logger := zap.New(zap.UseDevMode(true), zap.WriteTo(os.Stdout), zap.Encoder(logfmtEncoder))
+	logf.SetLogger(logger)
+```
+
+**NOTE**: For this example, you will need to add the module `"github.com/sykesm/zap-logfmt"` to your project. Run `go get -u github.com/sykesm/zap-logfmt`.
+
+#### Output using custom zap logger
+
+```console
+$ go run main.go
+ts=2020-04-30T20:35:59.551268Z level=info logger=global msg="Printing at INFO level"
+ts=2020-04-30T20:35:59.551314Z level=debug logger=global msg="Printing at DEBUG level"
+ts=2020-04-30T20:35:59.551318Z level=info logger=scoped msg="Printing at INFO level"
+ts=2020-04-30T20:35:59.55132Z level=debug logger=scoped msg="Printing at DEBUG level"
```

Comment:
Where the names `global ` and `scoped` was added above?

Reply:
yes, they are coming from below lines.
```Go
	globalLog.Info("Printing at INFO level")
	globalLog.V(1).Info("Printing at DEBUG level")
	scopedLog.Info("Printing at INFO level")
	scopedLog.V(1).Info("Printing at DEBUG level")
```

Reply:
@bharathi-tenneti I am unable to find the globalLog and scopedLog in the code of the specific example. It shows not be the output of this example. 

c/c @hasbro17

Reply:
@camilamacedo86 Can you please post the full example that you're testing so @bharathi-tenneti can verify the output that you're generating.

Reply:
![CASE 2](https://user-images.githubusercontent.com/56410941/81178432-2f422b80-8f76-11ea-8101-6843f83c8bfa.png)

Reply:
Your code in the IDEA is not == the example provided. See: https://github.com/operator-framework/operator-sdk/pull/2903/files/5aba3773b3ff4b4fbc8af5d721eecab42a24481c#diff-8316575205c0a990fd2aa963efe4730dR94-R115

Reply:
As discussed on slack, It is a running example, where you just replace CASE 1 with CASE 2, rest of the code block remains the same.